sdcm/nemesis.py

```diff
@@ -846,6 +846,55 @@ def _add_drop_column_run_in_cycle(self):
             self._add_drop_column()
             time.sleep(1)
 
+    def choose_partitions_for_delete(self, partitions_amount, ks_cf, with_clustering_key_data=False):
+        # This parameter is maximum -partition-count that data is written with -validate-data
+        max_partitions_with_data_validate = self.cluster.params.get('max_partitions_with_data_validate', 0)
+
+        partitions_for_delete = defaultdict(list)
+        with self.tester.cql_connection_patient(self.target_node) as session:
+            session.default_consistency_level = ConsistencyLevel.QUORUM
+
+            for _ in range(partitions_amount):
+                partition_key = random.randint(max_partitions_with_data_validate,
+                                               max_partitions_with_data_validate + 2000)
+                cmd = f"paging off; select ck from {ks_cf} where partition_key={partition_key}"
+                result = self.target_node.run_cqlsh(cmd=cmd, verbose=False, target_db_node=self.target_node,
+                                                    timeout=600, split=True, connect_timeout=60)
+                if not result:
+                    continue
+
+                if not with_clustering_key_data:
+                    partitions_for_delete[partition_key] = []
+                    continue
+
+                for row in result[4:]:
+                    if '|' not in row:
+                        continue
+                    clustering_key = int(row.split('|')[0].strip)
+                    partitions_for_delete[partition_key].append(clustering_key)
+
+        return partitions_for_delete
+
+    def distrupt_delete_by_partitions(self):
+        """
+        Delete few partitions in the table with large partitions
+        """
+        self._set_current_disruption('DeleteByPartitionsMonkey {}'.format(self.target_node))
+        test_keyspaces = self.cluster.get_test_keyspaces()
+
+        # if keyspace doesn't exist, create it by cassandra-stress
+        if 'scylla_bench' not in test_keyspaces:
+            raise UnsupportedNemesis("this nemesis can run on scylla_bench test")
+
+        ks_cf = 'scylla_bench.test'
+        partitions_for_delete = self.choose_partitions_for_delete(10, ks_cf)
+        partition_keys = ', '.join(str(pk) for pk in partitions_for_delete.keys())
+        self.log.debug(f'Partition keys for delete: {partition_keys}')
+
+        cmd = f"delete from {ks_cf} where pk in ({partition_keys})"
```

Comment:
Did you test it with very large partitions? (like the current large partition longevity?)
Need to make sure it doesn't timeout...
If it does, need to do it one by one.

Maybe also worth doing it inside helper and just calling it here with list of partitions to delete.

Reply:
Changed